File: src/main/kotlin/top/charjin/shoppingserver/mapper/OsUserMapper.java
package top.charjin.shoppingserver.mapper;

import org.apache.ibatis.annotations.Param;
import org.apache.ibatis.annotations.Select;
import top.charjin.shoppingserver.entity.OsUser;

public interface OsUserMapper {
    int deleteByPrimaryKey(Integer userId);

    int insert(OsUser record);

    int insertSelective(OsUser record);

    OsUser selectByPrimaryKey(Integer userId);

    int updateByPrimaryKeySelective(OsUser record);

    int updateByPrimaryKey(OsUser record);

    OsUser selectByUsernamePwd(@Param("username") String username, @Param("password") String password);

    int updateByUserNamePwdSelective(OsUser user);


    @Select("select count(*) from os_user where username = #{username}")
    int selectCountByUserName(@Param("username") String username);
}

File: src/main/kotlin/top/charjin/shoppingserver/mapper/OsUserMapper.kt
package top.charjin.shoppingserver.mapper

import org.apache.ibatis.annotations.Param
import org.apache.ibatis.annotations.Select
import top.charjin.shoppingserver.entity.OsUser

interface OsUserMapper {
    fun deleteByPrimaryKey(id: Int?): Int

    fun insert(record: OsUser): Int

    fun insertSelective(record: OsUser): Int

    fun selectByPrimaryKey(id: Int?): OsUser

    fun updateByPrimaryKeySelective(record: OsUser): Int

    fun updateByPrimaryKey(record: OsUser): Int

    @Select("select * from os_user where username = #{username} and password = #{password}")
    fun selectByUsernamePwd(@Param("username") username: String, @Param("password") password: String): OsUser

    fun updateByUserNamePwdSelective(user: OsUser): Int
}